libcxx/include/__config

```diff
@@ -350,6 +363,9 @@ _LIBCPP_HARDENING_MODE_DEBUG
 #    define _LIBCPP_ASSERT_PEDANTIC(expression, message)                 _LIBCPP_ASSUME(expression)
 #    define _LIBCPP_ASSERT_INTERNAL(expression, message)                 _LIBCPP_ASSUME(expression)
 #    define _LIBCPP_ASSERT_UNCATEGORIZED(expression, message)            _LIBCPP_ASSUME(expression)
+// TODO(hardening): if `_LIBCPP_ASSUME` becomes functional again (it's currently a no-op), this would essentially
+// discard the assumption.
+#    define _LIBCPP_REDUNDANT_ASSERTION(expression)                      ((void)0)
```

Comment:
```suggestion
#    define _LIBCPP_REDUNDANT_ASSERTION(assertion)                      ((void)0)
```

To make it clear that this is an assertion, not the boolean expression itself. This makes the usage of the macro clearer. And similarly below.